Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/zip.test.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+./zip.sh
+unzip ./Package/IncidentProducer.zip -d IncidentProducer.output1
+git checkout ./Package/IncidentProducer.zip
+unzip ./Package/IncidentProducer.zip -d IncidentProducer.output2
+kdiff3 IncidentProducer.output1 IncidentProducer.output2
+git clean -fd
```

Comment:
Will this affect the other untracked files that were not created as part of running this script? For example, if I added a new directory waiting to be merged(and independent of this script) and run this script, will this remove that directory?

Reply:
if your files not in git yet, and not ignored by .gitignore, then your files would be removed.

Reply:
Is that not an issue? Shouldn't we just clean what is created as part of this script so that our own workflows are not affected? In this case, should we just remove UNZIP_DIR?

Reply:
ennn....you should always keep your git working folder a health status, before running any scripts, right???

Reply:
generally, it is safe, only affect those not in git yet..

Reply:
if you want to protect your files not in git, and you really don't want to add them to git, then I would suggest ignore your files to .gitignore files.

Reply:
Why? Imagine that I'm in the middle of adding a new playbook directory and run this script for a different reason. This command will clean up all my work in the playbook directory.

It's usually clean if the script doesn't have pre-requisites or side effects.

Reply:
I could change this..
"in the middle of adding", you could git add, but not commit, then those files would not be affected..

Reply:
The user has to 'know' AND 'remember' that they should save the files before running the script. This approach is prone to errors. Also, isn't it a matter of simply deleting a UNZIP_DIR directory instead of git clean?

Reply:
that change was already done yesterday.
and it is just a test.sh, seldom used one.

Reply:
My bad. I'm not familiar with reviews in github.